File: src/prbm_greed/TrainingSuit.java
package prbm_greed;

public class TrainingSuit {

	public static void main(String[] args) {

	}
	
    public int solution(int n, int[] lost, int[] reserve) {
        int[] array = new int[n];
        for(int i : lost) array[i-1]--;
        for(int i : reserve) array[i-1]++;
        for(int i=0; i<array.length; i++) {
        	if(array[i]==1) {
        		// index 0ÀÏ ¶§
        		if(i==0) {
        			if(array[i+1]==-1) {
        				array[i]--;
						array[i+1]++;
        			}
        		} else if (i>0 && i<array.length-1) {
        			if(array[i-1]==-1) {
        				array[i]--;
						array[i-1]++;
        			} else if (array[i+1]==-1) {
        				array[i]--;
						array[i+1]++;        				
        			}        			
        		} else if (i==array.length-1) {
        			if(array[i-1]==-1) {
        				array[i]--;
						array[i-1]++;
        			}
        		}
        	}
        }
        int cnt=0;
        for(int i : array) {
        	if(i>-1) ++cnt;
        }
        return cnt;
    }
}

File: src/py/pg/prbm_greed/TrainingSuit.py
def solution(n, lost, reserve):
    answer = 0

    list = [1 for i in range(n)]
    for l in lost:
        list[l-1] -=1

    for r in reserve:
        list[r-1] +=1

    for idx, val in enumerate(list):
        if val == 2:
            if idx == 0:
                if list[idx+1]==0:
                    list[idx+1] +=1
                    list[idx] -=1
            elif idx == len(list)-1:
                if list[idx-1]==0:
                    list[idx-1] +=1
                    list[idx] -=1
            else:
                if list[idx-1]==0:
                    list[idx-1] +=1
                    list[idx] -=1
                elif list[idx+1]==0:
                    list[idx+1] +=1
                    list[idx] -=1

    for val in list:
        if val > 0:
            answer +=1                

    return answer